tools/validate-xep0001-conformance.sh

```diff
@@ -0,0 +1,205 @@
+#!/bin/bash
+
+# Checks if provided file (filename to be provided as first argument) conforms to XEP-0001.
+# See https://github.com/xsf/xeps/issues/1235
+#
+# Expected to be executed from the directory that holds all xep XML files.
+# Requires one argument: the file name of the xep to be validated, eg:
+#
+# $ tools/validate-xep0001-conformance.sh xep-0010.xml
+#
+# exit status will be non-zero upon validation failure.
+#
+# requires: bash, xmllint, sort (supporting the '-V' argument)
+
+set -euo pipefail
+
+validation_result=0
+
+# 1. Check DTD conformance against xep.dtd
+xep_dtd="xep.dtd"
+if xmllint --noout --dtdvalid "$xep_dtd" "$1"
+then
+  echo "[PASS] DTD conformance against xep.dtd"
+else
+  echo "[FAIL] DTD conformance against xep.dtd"
+  validation_result=1;
+fi
+
+file_name=$(basename -- "$1")
+
+# The test for exit code equalling 10 detects when the XPATH evaluation yields no results. In that case, the execution
+# should not fail immediately, but use an empty value instead (which will likely cause a validation failure further down).
+header_number=$(xmllint --xpath '/xep/header/number/text()' --nowarning --dtdvalid "$xep_dtd" "$1") || test $?=10
+header_status=$(xmllint --xpath '/xep/header/status/text()' --nowarning --dtdvalid "$xep_dtd" "$1") || test $?=10
```

Comment:
It might be sufficient running the `--dtdvalid` check only once. No need to run the entire document through several rounds of DTD checking.

Reply:
I agree that validation should be needed only once, but without it, I couldn't get xmllint to process the file. That's why I kept it in for the subsequent invocations.